build_projects/shared-build-targets-utils/Publishing/AzurePublisher.cs

```diff
@@ -103,12 +105,30 @@ public IEnumerable<string> ListBlobs(string virtualDirectory)
             return blobFiles.Results.Select(bf => bf.Uri.PathAndQuery);
         }
 
-        public string AcquireLeaseOnBlob(string blob)
+        public string AcquireLeaseOnBlob(string blob, int maxWaitSeconds=120000, int delayMilliseconds=500)
```

Comment:
`??` doesn't work on TimeSpans, Timespans can't be null, TimeSpan.Zero is not a compile time constant